File: src/main/java/com/zzkun/model/UHuntTreeNode.java
package com.zzkun.model;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

/**
 * Created by kun on 2016/7/5.
 */
public class UHuntTreeNode implements Comparable<UHuntTreeNode>, Serializable {

    public int deep;
    public int id;
    public String name;
    public String type;
    public List<UHuntTreeNode> son = new ArrayList<>();

    public UHuntTreeNode() {
    }

    public UHuntTreeNode(int deep, int id, String name, String type) {
        this.deep = deep;
        this.id = id;
        this.name = name;
        this.type = type;
    }

    @Override
    public String toString() {
        return "UHuntTreeNode{" +
                "deep=" + deep +
                ", id=" + id +
                ", name='" + name + '\'' +
                ", type='" + type + '\'' +
                '}';
    }

    @Override
    public int compareTo(UHuntTreeNode o) {
        return Integer.compare(id, o.id);
    }
}

File: src/main/java/com/zzkun/util/uhunt/UHuntTreeNode.kt
package com.zzkun.util.uhunt

import java.io.Serializable
import java.util.*

/**
 * Created by kun on 2016/7/5.
 */
class UHuntTreeNode : Comparable<UHuntTreeNode>, Serializable {

    var deep: Int = 0
    var id: Int = 0
    var name: String = ""
    var type: String = ""
    var son: List<UHuntTreeNode> = ArrayList()
    var pids: List<Int> = ArrayList()

    constructor() {
    }

    constructor(deep: Int, id: Int, name: String, type: String) {
        this.deep = deep
        this.id = id
        this.name = name
        this.type = type
    }



    override fun compareTo(other: UHuntTreeNode): Int {
        return Integer.compare(id, other.id)
    }

    override fun toString(): String {
        return "UHuntTreeNode(deep=$deep, id=$id, name='$name', type='$type', son=${son.size}, pids=${pids.size})"
    }

}
